### PR TITLE
Upload package files

### PR DESCRIPTION
**Pull request by darkkifir**
Add main files of package.

Best regards, Alex

### CODE REVIEW

File: Navidux/Package.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// swift-tools-version: 5.7
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Navidux",
+    platforms: [.iOS(.v13)],
+    products: [
+        // Products define the executables and libraries a package produces, and make them visible to other packages.
```
Comment (laqiguy) [inline Navidux/Package.swift:10]: There is no need in default comments in `Package.swift`

File: Navidux/.swiftpm/xcode/package.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (laqiguy) [inline Navidux/.swiftpm/xcode/package.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: I think you should add workspace in `.gitignore`

File: Navidux/Sources/Navidux/Coordinator.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import UIKit
+
+/// Main Object that used in navigation. Have many different types of presentation and inner methods to manipulate navigation stack.
+/// Work on states with associated values.
+/// - Note use **actionReducer(action:)** function with some action to change state of navigation stack. It's trigger inner function to change navigation screen.
+/// - Note use **start()** function to trigger first default screen or **actionReducer(action:)** with **restruct** action to set up inital navigation stack state.
+public protocol Coordinator: AnyObject {    
+    func actionReducer(action: Navigation.Action)
```
Comment (laqiguy) [inline Navidux/Sources/Navidux/Coordinator.swift:8]: I think there i some issues with codestyle. Mainly there should be new line after protocol declaration and first method/property.

File: Navidux/Sources/Navidux/NavigationController.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
+//
+//  NavigationController.swift
+//  NaviduxTests
+//
+//  Created by Александр Евсеев on 12.10.2022.
+//  Copyright © 2022 red_mad_robot. All rights reserved.
+//
```
Comment (laqiguy) [inline Navidux/Sources/Navidux/NavigationController.swift:7]: You need to decide do you need default comments for source files or not. Make it consistent😉

File: Navidux/Sources/Navidux/NavigationController.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import UIKit
+
+/// NavigationController its abstract object uses for navigation. Its a closest wrapper to UINavigation controller or alternative.
+/// - parameters:
+/// - **screens**: return some representation of screen from UINavigationController.viewController. Difference in track screen in navigation stack.
```
Comment (laqiguy) [inline Navidux/Sources/Navidux/NavigationController.swift:13]: You can do notation about properties and methods directly above them. No need to do it for whole protocol

File: Navidux/Sources/Navidux/implemention/NavigationRouter.swift [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            }
+            navigationController.pushViewController(screen, animated: animated)
+        case .modal:
+            screen.isModal = true
+            screen.navigationCallback = { [weak self, weak screen] in
+                self?.modalControllerDismissed(screenTag: screen?.tag)
+            }
+            state.hasOverlay = true
+            navigationController.present(screen, animated: animated, completion: nil)
+        // TODO: [JBNFS-26] Подготовить контейнер с боттом щитом
```
Comment (laqiguy) [inline Navidux/Sources/Navidux/implemention/NavigationRouter.swift:18]: Remove task number)

**Review by laqiguy — APPROVED** [review]
Fix comments and LGMT

File: Navidux/Sources/Navidux/Coordinator.swift (hunk 3 above)
Comment (darkkifir) [reply Navidux/Sources/Navidux/Coordinator.swift:8]: RMR-code-style don't have any appointments about protocols. In another 'Swift Style Guide' (https://google.github.io/swift/ from google) I found some examples with protocols. It's normal not to have new line after protocol declaration.

File: Navidux/.swiftpm/xcode/package.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 2 above)
Comment (darkkifir) [reply Navidux/.swiftpm/xcode/package.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: Added this file to gitignore

File: Navidux/Sources/Navidux/NavigationController.swift (hunk 4 above)
Comment (darkkifir) [reply Navidux/Sources/Navidux/NavigationController.swift]: remove unnecessary comments)

File: Navidux/Sources/Navidux/implemention/NavigationRouter.swift (hunk 6 above)
Comment (darkkifir) [reply Navidux/Sources/Navidux/implemention/NavigationRouter.swift]: remove tasks)

File: Navidux/Package.swift (hunk 1 above)
Comment (darkkifir) [reply Navidux/Package.swift]: remove comments

File: Navidux/Sources/Navidux/NavigationController.swift (hunk 5 above)
Comment (darkkifir) [reply Navidux/Sources/Navidux/NavigationController.swift]: move notations to properties where its necessary

File: Navidux/Sources/Navidux/Coordinator.swift (hunk 3 above)
Comment (laqiguy) [reply Navidux/Sources/Navidux/Coordinator.swift:8]: Oh, I'm ok with it then

**Pull request merged**